### PR TITLE
improving cluster-admins idp flow

### PR DESCRIPTION
**Pull request by zvikorn**
Signed-off-by: Tzvika Avni <tavni@redhat.com>

[SDA-6170](https://issues.redhat.com//browse/SDA-6170)

@pvasant

### CODE REVIEW

**Comment by zvikorn** [comment]
@pvasant @zgalor PTAL

**Comment by pvasant** [comment]
@zvikorn it should be cluster admin not cluster admins

File: cmd/create/admin/cmd.go [hunk 1]
```diff
@@ -186,6 +182,8 @@ func run(cmd *cobra.Command, _ []string) {
 	}
 
 	reporter.Infof("Admin account has been added to cluster '%s'.", clusterKey)
+	reporter.Infof("An Identity Provider named 'Cluster-Admins' from type HTPasswdIdentityProvider, " +
```
Comment (pvasant) [inline cmd/create/admin/cmd.go:185]: I would change it to I: An HTPasswd Identity Provider was created with the name 'Cluster-Admin' cc: @arendej

**Comment by pvasant** [comment]
Also there is another issue we need to handle. 

during rosa create admin flow , If the htpasswd idp fails we just create the admin and not the htpasswd so that causes SDA 6106 as well. I think the proper fix is to move the logic to cluser service. rosa would call create admin in CS and CS will handle checking for idp and add it if it is not available.

**Comment by arendej** [comment]
We need to take into account the issue in SDA-6170 as well, please

**Comment by zvikorn** [comment]
@pvasant Is it a final decision to move the entire logic to CS and remove unnecessary code from ROSA?

**Comment by pvasant** [comment]
> @pvasant Is it a final decision to move the entire logic to CS and remove unnecessary code from ROSA?

Yes we need to move the logic to CS  that way it is under same transaction and will be reverted if one of them fails 

`rosa would call create admin in CS and CS will handle checking for idp and add it if it is not available.`

**Comment by zvikorn** [comment]
Closing this in favour of https://github.com/openshift/rosa/pull/769

**Pull request closed without merging**